src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -919,6 +938,103 @@ internal override void TryForceResolve(bool asValueType)
             }
         }
 
+        /// <summary>
+        /// Extension for a type argument whose nullable annotation after substitution depends on the type constraints on the type parameter.
+        /// To avoid a cycle, we delay determining the nullable annotation until after the type parameter's type constraints are bound.
+        /// </summary>
+        private sealed class LazySubstitutedType : Extensions
+        {
+            private readonly ImmutableArray<CustomModifier> _customModifiers;
+            private readonly TypeParameterSymbol _typeParameter;
+            private int _resolved;
+
+            public LazySubstitutedType(ImmutableArray<CustomModifier> customModifiers, TypeParameterSymbol typeParameter)
+            {
+                Debug.Assert(!customModifiers.IsDefault);
+                _customModifiers = customModifiers;
+                _typeParameter = typeParameter;
+                _resolved = (int)NullableAnnotation.Ignored;
```

Comment:
Does the initial value of `_resolved` matter in any way? Consider using `-1` instead of a seemingly meaningful (but in fact not) value

Reply:
No, it's just a sentinel.